3_parameter_efficient_finetuning/lora_adapters.md

```diff
@@ -44,7 +74,11 @@ peft_model = PeftModel.from_pretrained(
 )
 
 # 3. Merge adapter weights with base model
-merged_model = peft_model.merge_and_unload()
+try:
+    merged_model = peft_model.merge_and_unload()
+except RuntimeError as e:
+    print(f"Merging failed: {e}")
+    # Implement fallback strategy or memory optimization
```

Comment:
Perhaps the try-except is not needed because we already get an error message from the SDK so we can just focus on the happy flow.